modules/activemq-artemis-launch/added/drain.sh

```diff
@@ -26,31 +27,69 @@ function waitForJolokia() {
 
 endpointsCode=$(curl -s -o /dev/null -w "%{http_code}" -G -k -H "${endpointsAuth}" ${endpointsUrl})
 if [ $endpointsCode -ne 200 ]; then
-  echo "Can't find endpoints with ips status <${endpointsCode}>" 
+  echo "Can't find endpoints with ips status <${endpointsCode}>"
   exit 1
 fi
 
 ENDPOINTS=$(curl -s -X GET -G -k -H "${endpointsAuth}" ${endpointsUrl}"endpoints/${ENDPOINT_NAME}")
+echo "=====got endpoints details:"
 echo $ENDPOINTS
+# we will find out a broker pod's fqdn name which is <pod-name>.<$HEADLESS_SVC_NAME>.<namespace>.svc.<domain-name>
+# https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/
 count=0
+foundTarget="false"
 while [ 1 ]; do
   ip=$(echo $ENDPOINTS | python2 -c "import sys, json; print json.load(sys.stdin)['subsets'][0]['addresses'][${count}]['ip']")
   if [ $? -ne 0 ]; then
     echo "Can't find ip to scale down to tried ${count} ips"
-    exit
+    exit 1
   fi
-
   echo "got ip ${ip} broker ip is ${BROKER_IP}"
   if [ "$ip" != "$BROKER_IP" ]; then
+    # found one, now get pod name
+    podName=$(echo $ENDPOINTS | python2 -c "import sys, json; print json.load(sys.stdin)['subsets'][0]['addresses'][${count}]['targetRef']['name']")
```

Comment:
Shouldn't we rather use python3 these days?

Reply:
well yes we should, I'll create a task. thx Roman.

Reply:
this 
 if [ "$ip" != "$BROKER_IP" ]; then
is not right, the $BROKER_IP is indeed the pod's host name so the comparison always yields true. It should compare the podName with $BROKER_IP.